src/Aspire.Cli/Commands/NewCommand.cs

```diff
@@ -243,15 +250,26 @@ public virtual async Task<string> PromptForOutputPath(string path, CancellationT
 
     public virtual async Task<string> PromptForProjectNameAsync(string defaultName, CancellationToken cancellationToken)
     {
-        return await interactionService.PromptForStringAsync(
+        var name = await interactionService.PromptForStringAsync(
             "Enter the project name:",
             defaultValue: defaultName,
-            validator: (name) => {
-                return ProjectNameValidator.IsProjectNameValid(name)
-                    ? ValidationResult.Success()
-                    : ValidationResult.Error("Invalid project name.");
+            validator: (name) =>
+            {
+                if (!ProjectNameValidator.IsProjectNameValid(name))
+                {
+                    var sanitizedName = ProjectNameValidator.SanitizeProjectName(name);
+                    interactionService.DisplayWarning($"Invalid project name '{name}'. It will be modified to '{sanitizedName}'.");
+                }
+                return ValidationResult.Success();
             },
             cancellationToken: cancellationToken);
+
+        if (!ProjectNameValidator.IsProjectNameValid(name))
+        {
+            name = ProjectNameValidator.SanitizeProjectName(name);
+        }
+
```

Comment:
I think this results in a confusing user experience. I spun this up and this is what it looked like:

![image](https://github.com/user-attachments/assets/2ea4bfa9-e5bc-4829-8f4c-83c38cb36d5f)

So I passed in an invalid name on the command line argument, so it prompted me to accept its sanitised version which I then typed in another (bad) replacement - which it then accepted.

I think we should change this code so that it santises the original input, but if they override it with another bad example it prompts again.